src/bytes.rs

```diff
@@ -1327,6 +1363,43 @@ impl BytesMut {
         self.reserve(extend.len());
         self.put_slice(extend);
     }
+
+    // TODO: This test fails because `into_vec` is private, but we want to keep this documentation
+    //       so I'm ignoring it. If this function is made public then remove the `ignore`.
+    /// Attempt to extract the underlying vec.
+    ///
+    /// If this object is backed by `Vec` and `Vec` is not shared
+    /// by other `Bytes` instances, that vec is returned.
+    ///
+    /// Otherwise new `Vec` is allocated and data is copied into it.
+    ///
+    /// Note that this `Bytes` data can start at non-zero offset
+    /// of the vector (e. g. after `split_to` call). In that case data
+    /// will be memmoved to the beginning of the vector before returning.
+    ///
+    /// This operation is `O(N)` in the worst case and `O(1)` in the best.
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
+    /// use bytes::BytesMut;
+    ///
+    /// let vec = vec![17, 19];
+    ///
+    /// // copy pointer for the test
+    /// let ptr = vec.as_slice().as_ptr();
+    ///
+    /// let vec = BytesMut::from(vec).into_vec();
+    ///
+    /// assert_eq!(vec![17, 19], vec);
+    ///
+    /// // memory is not allocated for the result, it is the same object
+    /// assert_eq!(ptr, vec.as_slice().as_ptr());
+    /// ```
+    fn into_vec(self) -> Vec<u8> {
```

Comment:
I am OK w/ making this public. It's fine to have inherent struct functions as well as `From` implementations.